src/command_helpers.rs

```diff
@@ -308,10 +309,29 @@ pub(crate) fn objects_from_files(files: &[Arc<Path>], dst: &Path) -> Result<Vec<
             })?
             .to_string_lossy();
 
+        // Function to find the position of the first occurrence of the target substring
+        fn find_target_position(s: &str, targets: &[&str]) -> Option<usize> {
+            let mut pos = None;
+            for target in targets {
+                if let Some(index) = s.find(target) {
+                     //If a target is found and pos is None, set it
+                         if pos.is_none() || index < pos.unwrap() {
+                             pos = Some(index);
+                         }
+                }
+            }
+            pos
+        }
+
+        let filtered_dirname = if let Some(pos) = find_target_position(&dirname, &target_substring) {
```

Comment:
IMHO stripping `rustc` isn't very reliable, user could have `rustc` somewhere in the path.

I think stripping `CARGO_MANIFEST_DIR` is a better solution.